openapi.yaml

```diff
@@ -937,317 +1058,539 @@ definitions:
+        $ref: '#/definitions/JobResource'
+  JobPostBody:
+    type: object
+    required:
+      - data
     properties:
-      id:
-        type: string
-        description: Meal plan ID
-      type:
-        type: string
-        example: meal-plans
-      links:
-          $ref: '#/definitions/SelfLink'
-      attributes:
+      data:
+        type: object
+        required:
+          - type
+          - attributes
         properties:
-          mealPlan:
-            type: string
-            description: Description of meal plan
-            example: Orange Rewards
-          balance:
-            type: number
-            format: float
-            description: Remaining balance on meal plan in USD
-            example: 23.8
-          lastUsedDate:
-            type: string
-            format: date-time
-            description: The last time the meal plan was used
-            example: '2017-07-14T16:45:15Z'
-          lastUsedLocation:
-            type: string
-            description: Description of place the meal plan was last used
-            example: 'Java II'
-  NewAddressResultObject:
+          type:
+            $ref: '#/definitions/JobType'
+          attributes:
+            allOf:
+              - $ref: '#/definitions/JobPostRequired'
+              - $ref: '#/definitions/JobAttributes'
+  JobPatchBody:
+    type: object
+    required:
+      - data
     properties:
       data:
-        $ref: '#/definitions/NewAddressResourceObject'
-  NewAddressResourceObject:
+        type: object
+        required:
+          - id
+          - type
+        additionalProperties: false
+        properties:
+          id:
+            $ref: '#/definitions/JobId'
+          type:
+            $ref: '#/definitions/JobType'
+          attributes:
+              $ref: '#/definitions/JobAttributes'
+  SsnPostAttributes:
+    type: object
+    required:
+      - ssn
+    additionalProperties: false
+    properties:
+      ssn:
+        type: string
+        description: 9 digits SSN
+        example: '123456789'
+  SsnPostBody:
+    type: object
+    required:
+      - data
+    additionalProperties: false
+    properties:
+      data:
+        type: object
+        required:
+          - type
+          - attributes
+        additionalProperties: false
+        properties:
+          type:
+            $ref: '#/definitions/SsnType'
+          attributes:
+            $ref: '#/definitions/SsnPostAttributes'
+  MealPlanAttributes:
+    type: object
+    properties:
+      mealPlan:
+        type: string
+        description: Description of meal plan
+        example: Orange Rewards
+      balance:
+        type: number
+        format: float
+        description: Remaining balance on meal plan in USD.
+        example: 3.50
+      lastUsedDateTime:
+        type: string
+        format: date-time
+        description: The last time the meal plan was used.
+        example: '2017-07-14T16:45:15Z'
+      lastUsedLocation:
```

Comment:
well... I'm not sure what should be combined and what should not now lol.

```json
{
    "lastUsed": {
        "dateTime": "2017-07-14T16:45:15Z",
        "location": "Java II"
    }
}
```

Reply:
Yeah we should be careful but I think it works here.